### PR TITLE
[5.8] fixed required_with with wildcard

### PR DESCRIPTION
**Pull request by mx2s**
reopened: #28813
issue: #26957

required_with didn't work with wildcard, see source issue for more details

### CODE REVIEW

**Comment by devcircus** [comment]
Just a suggestion, but it helps the maintainers if you you include a full description of the PR instead of just linking issues. Show the problem, what happens currently, what will happen if the pr is accepted, what you changed to make the new behavior possible, and why the new way is better.

File: src/Illuminate/Validation/Concerns/ValidatesAttributes.php [hunk 1]
```diff
@@ -1391,7 +1391,7 @@ public function validateRequiredUnless($attribute, $value, $parameters)
      */
     public function validateRequiredWith($attribute, $value, $parameters)
     {
-        if (! $this->allFailingRequired($parameters)) {
+        if (strpos($parameters[0], '*') || ! $this->allFailingRequired($parameters)) {
```
Comment (bonzai) [inline src/Illuminate/Validation/Concerns/ValidatesAttributes.php]:
```suggestion
        if (strpos($parameters[0], '*') !== false || ! $this->allFailingRequired($parameters)) {
```

**Review by themsaid — CHANGES_REQUESTED** [review]
This test will not pass with your changes:

```
$v = new Validator(
    $trans,
    [
        'names' => [['second' => 'Otwell']],
        'foo' => '',
    ],
    ['foo' => 'required_with:names.*.first']
);
$this->assertTrue($v->passes());
```

**Comment by mxss** [comment]
@themsaid I've updated PR (added your test + addional check), please review

**Comment by themsaid** [comment]
Still won't work if value is `null` or the key is not even provided.

I don't really think this can be fixed with a simple solution, `getValue()` won't accept a wildcard attribute name so it'll always return `null` causing whatever you provide in `required_with` rule to pass without validation.

I'm closing this PR since it's not right and breaking other validation cases. Please feel free to open a new one if you managed to get `getValue` to work with wildcard attributes.

**Pull request closed without merging**